provider/core/src/hello_world.rs

```diff
@@ -41,6 +42,21 @@ impl Default for HelloWorld<'_> {
     }
 }
 
+impl MaybeAsVarULE for HelloWorld<'_> {
```

Comment:
Issue: I think allowing implementation of this trait on arbitrary data structs is going to be super confusing (and it should definitely not use `From` to do the reverse).

`HelloWorldV1::DataStruct` should just be `Cow<'static, str>`, and this trait should be implemented for `Cow<'static, str>` in `icu_provider`.

Reply:
I'm okay implementing it on `Cow`, but we still need the no-op version of the trait implement on other things. I don't see a reason to _not_ allow it to be implemented on arbitrary data structs.

Reply:
(I am still allowing the trait to be implemented on any data struct)